src/components/UserContainer.tsx

```diff
@@ -1,48 +1,68 @@
-import React from 'react';
+import React, { Component } from 'react';
 import './UserContainer.css';
 import * as Types from '../constants/Types';
 
 type UserContainerProps = {
     displayName: Types.UserDisplay;
-    onClick: (displayName: Types.UserDisplay) => void;
+    onClick?: () => void;
+    onLeaveRoom?: () => void;
     requestSent?: boolean;
     accepted?: boolean;
     currentRoom?: boolean;
 };
 
-const UserContainer: React.FC<UserContainerProps> = ({
-    displayName,
-    onClick,
-    requestSent,
-    accepted,
-    currentRoom,
-}: UserContainerProps) => {
-    const currentRoomNoAction = !accepted && !requestSent && currentRoom;
-    let status = <p></p>;
-    if (accepted) {
-        status = <p className="user-display-status">Accepted</p>;
-    } else {
-        status = requestSent ? <p className="user-display-status">Pending</p> : <p></p>;
-    }
+type UserContainerState = {
+    hovered: boolean;
+}
 
-    return (
-        <div
-            className={`user-container ${currentRoom && 'user-container-current'} ${currentRoomNoAction &&
-                'user-container-initial'}`}
-            onMouseDown={(): void => {
-                onClick(displayName);
-            }}
-        >
-            <div
-                className={`user-display-icon ${currentRoomNoAction && 'user-display-light'}`}
-                style={{ backgroundColor: displayName.color }}
-            ></div>
-            <p className={`user-display-text ${currentRoomNoAction && 'user-display-light'}`}>
-                {displayName.displayName}
-            </p>
-            {status}
-        </div>
-    );
-};
+export default class UserContainer extends Component<UserContainerProps, UserContainerState> {
+    state: UserContainerState = {
+        hovered: false
+    };
 
-export default UserContainer;
+    render(): React.ReactNode {
+        const currentRoomNoAction = !this.props.accepted && !this.props.requestSent && this.props.currentRoom;
+        let status = <p></p>;
```

Comment:
Would this be better if it was something like this?

```
const acceptedStatusHtmlContent = <p className="user-display-status">Accepted</p>;
const pendingStatusHtmlContent = <p className="user-display-status">Pending</p>;
const status = this.props.accepted ? acceptedStatusHtmlContent : pendingStatusHtmlContent;
```

Reply:
I like this!